build-caching/src/main/java/com/vertispan/j2cl/build/TranslationsFileConfiguration.java

```diff
@@ -0,0 +1,102 @@
+package com.vertispan.j2cl.build;
+
+import com.vertispan.j2cl.build.task.CachedPath;
+import com.vertispan.j2cl.build.task.Config;
+import com.vertispan.j2cl.build.task.Input;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class TranslationsFileConfiguration {
+
+    private boolean auto;
+
+    private File file;
+
+    private Config config;
+    List<com.vertispan.j2cl.build.task.Input> jsSources;
+
+    public TranslationsFileConfiguration(Config config, File file) {
```

Comment:
in this ctor, `auto` is effectively false, right? maybe make `file` and `auto` final, assign them both? or make `file` an optional, so that there is just one field to hold that?